### PR TITLE
Loader for .mtr tunes (Master Tracker)

### PR DESCRIPTION
**Pull request by dmitrysmagin**
Hi.
I've noted that mtr is still not supported: http://adplug.github.io/library/entry/ArkhamMasterTracker24.html
so I've added a loader which reuses protrack.cpp.

Notes:
- Original AKMTRK2.DOC has lots of errors and inconsistencies on file format, so some guess was applied.
- Almost all commands are translated except for retrigger and order jumps, but I didn't find any tunes that use them.
- Most tunes sound very close compared with the original tracker, but some tunes (especially converted from MODs) have ridiculous out-of-tune slides. That's not a player's error, the tunes are composed like that.

### CODE REVIEW

**Comment by mywave82** [comment]
I will start looking at this PR sometime after the weekend

File: src/mtr.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    char header[51] = {0};
+    char mtitle[21] = {0};
+    int nvoices, ndigvoices, npatterns, orderlen,
+        timervalue = 0x428F, restart, flength;
+
+    f->readString(header, 50);
+
+    if (!strncmp(header, "MTRAC ", 6)) {
+        version = 1;
+        sscanf(header + 26, "%02x %02x %02x %02x %08x",
```
Comment (mywave82) [inline src/mtr.cpp:50]: Need to error-out if sscan() returns another value than 5

(Which will be the case for .MTR (v1.x) files)

File: src/mtr.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            &orderlen,
+            &restart,
+            &flength
+        );
+        strncpy(mtitle, header + 6, 20);
+        timervalue = f->readInt(2); // usually 428F
+        f->ignore(1); // 0=SPK 1=ADL 2=SBP
+    } else if (!strncmp(header, "MTRACK NC", 9)) {
+        version = 2;
+        sscanf(header + 10, "%02x %02x %02x %02x %02x %02x %04x %08x",
```
Comment (mywave82) [inline src/mtr.cpp:62]: Need to error-out if sscan() returns another value than 8

**Review by mywave82 — CHANGES_REQUESTED** [review]

File: src/mtr.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    ninstruments = 0;
+
+    char header[51] = {0};
+    char mtitle[21] = {0};
+    int nvoices, ndigvoices, npatterns, orderlen,
+        timervalue = 0x428F, restart, flength;
+
+    f->readString(header, 50);
+
+    if (!strncmp(header, "MTRAC ", 6)) {
```
Comment (mywave82) [inline src/mtr.cpp:48]: Documentation says "MTRACK ". Is the documentation incorrect, if so, add a comment please

**Review by mywave82 — CHANGES_REQUESTED** [review]

File: src/mtr.cpp [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    } else if (!strncmp(header, "MTRACK NC", 9)) {
+        version = 2;
+        sscanf(header + 10, "%02x %02x %02x %02x %02x %02x %04x %08x",
+            &nvoices,
+            &ndigvoices, // unused
+            &npatterns,
+            &orderlen,
+            &ninstruments,
+            &restart,
+            &timervalue, // usually 428F
```
Comment (mywave82) [inline src/mtr.cpp:69]: timervalue also seem to be a deviation from the documentation, please add comment :-)

**Review by mywave82 — CHANGES_REQUESTED** [review]

File: src/mtr.cpp (hunk 1 above)
Comment (dmitrysmagin) [reply src/mtr.cpp:50]: v1 files have "MTRACK\x20" at the beginning, v2 have "MTRACK\x20NC\x20".
sscanf shouldn't fail once the version is recognized.
AKMTRK2.DOC is incorrect describing this. There are lots of errors there, too much to enumerate them all, easier to create another 'fixed' variant of a doc.

File: src/mtr.cpp (hunk 3 above)
Comment (dmitrysmagin) [reply src/mtr.cpp:48]: It is incorrect.

File: src/mtr.cpp (hunk 4 above)
Comment (dmitrysmagin) [reply src/mtr.cpp:69]: Yep, in v1 it's a binary value after the string header, in v2 it's one of the hex values in the header string. Obviously, this is a timer value written into timer ports, so hz is recalculated back as 0x1234DD / timervalue

File: src/mtr.cpp (hunk 2 above)
Comment (dmitrysmagin) [reply src/mtr.cpp:62]: Okay, some error handling for sscanf won't hurt if the file is severely broken.

File: src/mtr.cpp (hunk 3 above)
Comment (mywave82) [reply src/mtr.cpp:48]: We might get lucky to find a user with a rare file from a version that is not in the modland.com archives

**Comment by dmitrysmagin** [comment]
Added error checking on sscanf().
I've tested all .MTR files from the archive with the Master Tracker, all seem to be recognized fine.
If there are others somewhere, I'll be willing to test them.

File: src/mtr.cpp (hunk 1 above)
Comment (dmitrysmagin) [reply src/mtr.cpp]: Fixed.

File: src/mtr.cpp (hunk 3 above)
Comment (dmitrysmagin) [reply src/mtr.cpp:51]: I've checked http://modland.com/pub/modules/Ad%20Lib/Master%20Tracker/ and it contains the same MTR modules that were packed with the original Master Tracker. I've tested them all with this loader, and they play identically with MT.

**Comment by mywave82** [comment]
Thank you for your work and patience

**Pull request merged**